x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -116,21 +118,32 @@ public String toString() {
         }
     }
 
-    public static class TaskParams implements PersistentTaskParams {
+    public static class TaskParams implements PersistentTaskParams, MlTaskParams {
 
         public static final Version VERSION_INTRODUCED = Version.V_8_0_0;
 
+        private static final ParseField MODEL_BYTES = new ParseField("model_bytes");
+
+        /**
+         * This has been found to be approximately 300MB on linux by manual testing.
+         * TODO Check if it is substantially different in other platforms.
+         */
+        private static final ByteSizeValue MEMORY_OVERHEAD = ByteSizeValue.ofMb(300);
```

Comment:
`MachineLearning.NATIVE_EXECUTABLE_CODE_OVERHEAD` (currently 30MB) will also be added on, so it might be worth deducting it from here.

I imagine more than 30MB of this 300MB is from the PyTorch code, so will only be incurred once per VM.  It would be hard to fully model that, but at least we can treat 30MB of it as being down to shared code.